Remove ENForm block

comzeradd opened the pull request:
Following its retirement from the theme code.

This also removes the Counter block to avoid cleaning it up from ENForm related code.

---

Ref: https://jira.greenpeace.org/browse/PLANET-7673

### Testing
- This should be tested along side with https://github.com/greenpeace/planet4-master-theme/pull/2466 
- The only thing that this PR may affect and should be tested is the Counter block

mardelnet reviewed (commented):
I am not sure about this: should we remove the `Fields_Model` class, too?

comzeradd commented on planet4-gutenberg-blocks.php:
Yes, good catch. I removed that too

comzeradd commented:
> I am not sure about this: should we remove the `Fields_Model` class, too?

Yes, those are all ENForm related. I had already removed the code that uses the Models classes, so I just removed those too.
